rules/static_framework_generator.bzl

```diff
@@ -0,0 +1,214 @@
+
+    fail("The target %s does not produce any framework" % target.label.name)
+
+def _get_existing_framework_name(target):
+    existing_framework_files = target.files.to_list()
+    if existing_framework_files:
+        subpaths = existing_framework_files[0].dirname.split("/")
+        if subpaths:
+            framework_name = subpaths[-1]
+            if framework_name.endswith(".framework"):
+                return framework_name.replace(".framework", "")
+
+    fail("The target %s does not produce any framework" % target.label.name)
+
+def _declare_framework(ctx, target):
+    framework_name = _get_existing_framework_name(target)
+    return _declare_directory(ctx, framework_name, framework_name + ".framework")
+
+def _make_frameworks(ctx):
+    frameworks = []
+    for target in ctx.attr.targets:
+        framework = _declare_framework(ctx, target)
+        resource_bundles = _declare_resource_bundles(ctx, target)
+
+        existing_framework_files = target.files.to_list()
+        existing_framework_path = _get_existing_framework_path(target)
+
+        resource_paths = " ".join([bundle.path for bundle in resource_bundles])
+        ctx.actions.run_shell(
+            outputs = [framework],
+            inputs = existing_framework_files + resource_bundles,
+            command = 'rsync -a "$1/" "$3"; for resource in $2; do rsync -a "$resource" "$3/Resources"; done',
+            arguments = [existing_framework_path, resource_paths, framework.path],
+        )
+        frameworks.append(framework)
+    return frameworks
+
+def _make_destination(ctx):
+    if ctx.attr.destination_relative_to_package:
+        return ctx.label.package + "/" + ctx.attr.destination_relative_to_package
+    else:
+        return ctx.attr.destination_relative_to_workspace
+
+def _make_installer(ctx, subpath, files, destination):
+    installer = ctx.actions.declare_file(subpath + "/installer.sh")
+    ctx.actions.expand_template(
+        template = ctx.file._installer_template,
+        output = installer,
+        substitutions = {
+            "$(files)": " ".join(files),
+            "$(destination)": destination,
+        },
+        is_executable = True,
+    )
+    return installer
+
+def _static_framework_generator_impl(ctx):
+    frameworks = _make_frameworks(ctx)
+    destination = _make_destination(ctx)
+    installer = _make_installer(ctx, ctx.label.name, [framework.path for framework in frameworks], destination)
+    return [
+        DefaultInfo(
+            executable = installer,
+            runfiles = ctx.runfiles(files = frameworks),
+        ),
+    ]
+
+# Making below rule public in order to properly generate its documentation until the
+# https://github.com/bazelbuild/stardoc/issues/27 issue is resolved
+static_framework_generator_test = rule(
+    implementation = _static_framework_generator_impl,
+    doc = """\
+This generator moves the frameworks artifacts produced by Bazel to a location out of the cache.
```

Comment:
how is this any different from `bazel build && cp bazel-bin/...` ? I guess this description doesn't explain to my why this rule needs to exist

Reply:
When running `bazelisk build --apple_platform_type=ios --deleted_packages='' tests/ios/static_framework_generator:TargetXIB` nothing gets created under `bazel-bin`. What gets created is:
* `bazel-out/applebin_ios-ios_x86_64-dbg-ST-78e822f89cdf/bin/tests/ios/static_framework_generator/TargetXIB/TargetXIB.framework`
* I would expect to see the resource bundles also created somewhere, but I cant find them anywhere. Seems like they are not generated when doing `build`

When you `cp` from inside `bazel-bin` or `bazel-out`, it is difficult to reliably tell where the `*.framework` and the resource bundles are located inside `bazel-bin` or `bazel-out`.

This rule solves those issues, serving a fully working framework in a location specified by the consumer